### DISCUSSION TITLE
Does my OIDC configuration work?

### DISCUSSION THREAD

**Discussion by rolego** [saleor/saleor#9568] [Q&A] [answered]
Dear all
I'm excited that the OID-plugin is now available in the cloud and thus I started to integrate it with our Auth0 instance.

After configuring everything I queried "me"  and got "null" as result:
`query {
  me {
    email
    id
  }
}
`
`
{
  "data": {
    "me": null
  },
  "extensions": {
    "cost": {
      "requestedQueryCost": 1,
      "maximumAvailable": 50000
    }
  }
}
`

I'm now asking myself: 
Am I supposed to get a result to this query? ...or what is the customer lifecycle when using OIDC?
Is there a problem in my OIDC configuration? How could I debug / analyse this?

Thanks for any hint
Roland

**Comment by rolego** [comment, marked as answer]
I'm meanwhile a step further: my email and other infos are returned by the "me" query.
The problem was that the User Info endpoint was missing in the audience of the access token.

**Reply by korycins** [reply]
> Now I'm faced with another problem: when authenticating with OIDC the users email is not confirmed.

Is not confirmed on Saleor side or OAuth provider?

if on Saleor side, then keep in mind that when you're using OIDC Plugin, there is no need to use Saleor's mutations related to the account registration. Any user received from OAuth provider is automatically active on Saleor side.

> As far as i understood, it's not possible to switch email confirmation off. Apparently it's also not possible to confirm the email with the API directly.

On Saleor side? If yes, then you can disable an account confirmation for Saleor default account registration flow (`settings.ENABLE_ACCOUNT_CONFIRMATION_BY_EMAIL`). But if you use OIDC plugin, this should not have any impact on you, as we always create an active user for the OIDC flow.

>Because of this, the customer profile (e.g. order) are not linked to the appropriate user.

Linking anonymous orders to newly created customers by OIDC Plugin is missing. If this is something that you need, then please create an issue for this.

**Reply by rolego** [reply]
Dear @korycins 
Thanks for your answer. I meanwhile (at the same time you wrote this response) created a dedicated issue for this problem (https://github.com/saleor/saleor/discussions/9577#discussion-4024238)
-> I will link your answer in the other issue. 
Have a nice day.

**Reply by rolego** [reply]
issue created: https://github.com/saleor/saleor/issues/9579
